pkg/ccl/backupccl/restore_planning.go

```diff
@@ -2003,9 +2003,14 @@ func doRestorePlan(
 		types = append(types, desc)
 	}
 
+	overrideDBName := intoDB
```

Comment:
could you add an assertion failure here, where we fail if intoDB is non-empty and newDBName is also non empty? intoDB is a table restore operation and newDBName is database level restore operation.

Reply:
I like this idea but I want to do it on master as well.  Will do it in a follow-up PR that we backport.